ui/vic-ui-h5c/vic/src/vic-app/src/app/services/data-property.service.ts

```diff
@@ -15,39 +15,53 @@
 */
 
 import { Injectable } from '@angular/core';
-import { Observable, BehaviorSubject } from 'rxjs/Rx';
+import { Observable, Subject } from 'rxjs/Rx';
 import { Http, Response } from '@angular/http';
 
+import { GlobalsService } from '../shared/globals.service';
+import { APP_CONFIG } from '../shared/app-config';
+
 import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/catch';
 
 import {
-    VirtualMachine,
     CONTAINER_VM_IMAGE_NAME_KEY,
     CONTAINER_VM_PORTMAPPING_KEY,
     CONTAINER_PRETTY_NAME_KEY,
     VCH_VM_CLIENT_IP_KEY,
     VCH_VM_ENDPOINT_PORT,
     VCH_VM_LOG_PORT
-} from '../vm.interface';
+} from '../vm.constants';
+import { VirtualMachine } from '../vm.interface';
+import { getVmStubData } from './vmStub';
 
 @Injectable()
 export class DataPropertyService {
-    private _results: BehaviorSubject<VirtualMachine> = new BehaviorSubject<VirtualMachine>(null);
-
-    constructor(private http: Http) {}
+    private _objectId: string;
+    private vmInfoSource: Subject<VirtualMachine> = new Subject<VirtualMachine>();
+    private vicObjectSource: Subject<any> = new Subject<any>();
+    public vmInfo$: Observable<VirtualMachine> = this.vmInfoSource.asObservable();
+    public vicObject$: Observable<any> = this.vicObjectSource.asObservable();
+
+    constructor(
+        private http: Http,
+        private gs: GlobalsService
+    ) {
+        this._objectId = this.gs.getWebPlatform().getObjectId();
+    }
 
```

Comment:
This file doesn't have any comments. Make sure to add comments to this file and any others.